corelibs do not build on gcc

```
/home/vic/work/hexapod_project/codebase/prj.tests/leg_test/src/leg_test.cpp: In function ‘int main(int, char**)’:
/home/vic/work/hexapod_project/codebase/prj.tests/leg_test/src/leg_test.cpp:36:367: error: no matching function for call to ‘Leg::forwardKinematics(Eigen::Vector3d)’
     Eigen::Vector3d traj = testLeg.forwardKinematics(Vector3d(0.04*sin(i*dt*(EIGEN_PI*2/3)), 0.1 + 0.1*sin(i*dt*(EIGEN_PI*2/5)), -0.1 + 0.2*sin(i*dt*(EIGEN_PI*2/5))));
                                                                                                                                                                                                                                                                                                                                                                               ^
/home/vic/work/hexapod_project/codebase/prj.tests/leg_test/src/leg_test.cpp:36:367: note: candidate is:
In file included from /home/vic/work/hexapod_project/codebase/prj.tests/leg_test/src/leg_test.cpp:4:0:
/home/vic/work/hexapod_project/codebase/prj.corelibs/leg/leg.h:34:19: note: Eigen::Vector3d Leg::forwardKinematics(Eigen::Vector3d&)
   Eigen::Vector3d forwardKinematics(Vector3d& targetAngles);
                   ^
/home/vic/work/hexapod_project/codebase/prj.corelibs/leg/leg.h:34:19: note:   no known conversion for argument 1 from ‘Eigen::Vector3d {aka Eigen::Matrix<double, 3, 1>}’ to ‘Eigen::Vector3d& {aka Eigen::Matrix<double, 3, 1>&}’
make[2]: *** [prj.tests/leg_test/CMakeFiles/leg_test.dir/src/leg_test.cpp.o] Error 1
make[1]: *** [prj.tests/leg_test/CMakeFiles/leg_test.dir/all] Error 2
make: *** [all] Error 2

```

calling functions with non-constant reference arguments with temporary objects seems to be the issue. Either make references constant or remove ref-quailifers from functions

--- a/codebase/prj.corelibs/joint/src/joint.cpp
+++ b/codebase/prj.corelibs/joint/src/joint.cpp
@@ -4,7 +4,7 @@ Joint::Joint():
   currentState(0)
 {
   this->pidControl.push_back(PID(0,0,0));
-  this->type = JOINT_TYPE::TRANSLATION_1D;
+  this->type = TRANSLATION_1D; //prun: either use enum class or don't use enum name::as a namespace
 }
 
 Joint::~Joint()

--- a/codebase/prj.corelibs/leg/leg.h
+++ b/codebase/prj.corelibs/leg/leg.h
@@ -30,10 +30,10 @@ public:
  bool init(const std::string name, const vector<double>& segments, const Eigen::Vector3d& mountingPoint, const Eigen::Vector3d& mountingAngles);
 public:
   vector<double> segments;
-  Eigen::Vector3d inverseKinematics(Vector3d& targetPoint);
-  Eigen::Vector3d forwardKinematics(Vector3d& targetAngles);
-  bool checkReachability(Vector3d& targetPoint);
-  bool recieveFB(VectorXd& feedback, int index);
+  Eigen::Vector3d inverseKinematics(const Vector3d& targetPoint);
+  Eigen::Vector3d forwardKinematics(const Vector3d& targetAngles);
+  bool checkReachability(const Vector3d& targetPoint);
+  bool recieveFB(const VectorXd& feedback, int index);
   bool setCoeffs(const PID::PIDcoeffs &coeffs);
 
 

--- a/codebase/prj.corelibs/leg/src/leg.cpp
+++ b/codebase/prj.corelibs/leg/src/leg.cpp
@@ -41,7 +41,7 @@ bool Leg::init(const std::string name, const Eigen::Vector3d& segments, const Ve
   return this->init(name, segms, mountingPoint ,mountingAngles);
 }
 
-Eigen::Vector3d Leg::inverseKinematics(Eigen::Vector3d& targetPoint)
+Eigen::Vector3d Leg::inverseKinematics(const Eigen::Vector3d& targetPoint)
 {
   double p1 = this->segments[0];
   double p2 = this->segments[1];
@@ -84,7 +84,7 @@ Eigen::Vector3d Leg::inverseKinematics(Eigen::Vector3d& targetPoint)
 }
 
 
-Eigen::Vector3d Leg::forwardKinematics(Eigen::Vector3d& targetAngles)
+Eigen::Vector3d Leg::forwardKinematics(const Eigen::Vector3d& targetAngles)
 {
   Eigen::Vector3d solution;
 
@@ -108,7 +108,7 @@ Eigen::Vector3d Leg::forwardKinematics(Eigen::Vector3d& targetAngles)
 }
 
 
-bool Leg::checkReachability(Vector3d& targetPoint)
+bool Leg::checkReachability(const Vector3d& targetPoint)
 {
   Eigen::Vector3d segm(segments[0], segments[1], segments[2]);
 
@@ -136,7 +136,7 @@ Eigen::Vector3d Leg::trajectoryGenerator(double time)
 
 }
 
-bool Leg::recieveFB(VectorXd& feedback, int index)
+bool Leg::recieveFB(const VectorXd& feedback, int index)
 {
   this->FBcoords = feedback.segment<3>(index);
   this->FBvelocities = feedback.segment<3>(index + 3);

--- a/codebase/prj.corelibs/trajectory/src/trajectory.cpp
+++ b/codebase/prj.corelibs/trajectory/src/trajectory.cpp
@@ -1,4 +1,4 @@
-#include <Trajectory/Trajectory.h>
+#include <trajectory/trajectory.h> //prun: unix is case-sensitive
 
 
 Trajectory::Trajectory()

--- a/codebase/prj.tests/joint_test/src/joint_test.cpp
+++ b/codebase/prj.tests/joint_test/src/joint_test.cpp
@@ -17,7 +17,7 @@ int main(int argc, char** argv)
 
   PID::PIDcoeffs coeffs={1,0,1};
 
-  joint.init("my joint", 0, Joint::JOINT_TYPE::ROTATION_3D, coeffs, &parent, &child);
+  joint.init("my joint", 0, Joint::ROTATION_3D, coeffs, &parent, &child);
   
   VectorXd state(joint.getDOFnumber());
   state.fill(0);

